sql/patches/patch40-userdomains.sql

```diff
@@ -0,0 +1,66 @@
+# noinspection SqlResolveForFile
+
+DROP PROCEDURE IF EXISTS SCHEMA_UPGRADE_SCRIPT;
+DELIMITER ';;'
+CREATE PROCEDURE SCHEMA_UPGRADE_SCRIPT() BEGIN
+    -- -------------------------------------------------------------------------
+    -- Developers - set the number of the schema patch here!
+    -- -------------------------------------------------------------------------
+    DECLARE patchversion INT DEFAULT 40;
+    -- -------------------------------------------------------------------------
+    -- working variables
+    DECLARE currentschemaversion INT DEFAULT 0;
+    DECLARE lastversion INT;
+	
+    -- check the schema has a version table
+    IF NOT EXISTS (SELECT * FROM information_schema.tables WHERE table_name = 'schemaversion' AND table_schema = DATABASE()) THEN
+        SIGNAL SQLSTATE '45000' SET message_text = 'Please ensure patches are run in order! This database does not have a schemaversion table.';
+    END IF;
+    
+    -- get the current version
+    SELECT version INTO currentschemaversion FROM schemaversion;
+    
+    -- check schema is not ahead of this patch
+    IF currentschemaversion >= patchversion THEN
+        SIGNAL SQLSTATE '45000' SET message_text = 'This patch has already been applied!';
+    END IF;
+    
+    -- check schema is up-to-date
+    SET lastversion = patchversion - 1;
+    IF currentschemaversion != lastversion THEN
+		SET @message_text = CONCAT('Please ensure patches are run in order! This patch upgrades to version ', patchversion, ', but the database is not version ', lastversion);
+        SIGNAL SQLSTATE '45000' SET message_text = @message_text;
+    END IF;
+
+    -- -------------------------------------------------------------------------
+    -- Developers - put your upgrade statements here!
+    -- -------------------------------------------------------------------------
+
+    alter table domain
+        add localdocumentation varchar(255) null;
+
+    -- Yes, I do want to update every row.
+    -- noinspection SqlConstantCondition,SqlConstantExpression
+    update domain set localdocumentation = 'https://en.wikipedia.org/wiki/Wikipedia:Request_an_account/Guide' where 1=1;
+
+    alter table domain
+        modify localdocumentation varchar(255) not null;
+
+    -- Add every user to every domain. In prod, this will be exactly one domain for now.
+    -- noinspection SqlWithoutWhere
+    delete from userdomain;
+
+    insert into userdomain (user, domain)
+    select u.id user, d.id domain
+    from user u cross join domain d;
+
+    commit;
```

Comment:
I don't see a transaction corresponding to this commit; did you intend to start one?

Reply:
No; this is mostly habit for me from Oracle which has autocommit disabled by default. Putting an explicit commit in also covers the situation in case someone's mysql client has autocommit disabled too